### ISSUE TITLE
undocked windows: No focus when selected and window selection isnt always marked 

### ISSUE THREAD

**Issue by un1versal** [kvirc/KVIrc#2109]
this is part 2 of https://github.com/kvirc/KVIrc/issues/2108
### Expected behavior

things that sound logical:

1) that no matter where you click over channel name that this should select the undocked window.
2) that when selecting and undocked window (thats in background somewhere) is brought forward and given focus.
3) be able to navigate with mouse / keyboard
4) that when right clicking an undocked window you have a visual confirmation the window is selected.
### Actual behavior

1) only when selecting on the **left outer-side** you can actually select and undocked window
2) no matter if window is selected or not, you dont ever get window brought forward or given any focus.
3) only mouse navigation partly works, keyboard nav is interrupted when reaching undocked window.
4) no visual confirmation hence performing actions is somewhat you hope its the right window.
### Steps to reproduce the issue

Here I undock window and click around on inside and outside trying to get focus and window selected / or right clicking...

![demo-ab](https://cloud.githubusercontent.com/assets/3521959/17653183/e602b0ca-6288-11e6-9939-39df2b50623c.gif)
### System information

```
KVIrc 4.9.2 'Aria'

Runtime Info:
System name: Linux 4.4.0-34-generic
System version: #53-Ubuntu SMP Wed Jul 27 16:06:39 UTC 2016
Architecture: x86_64
Qt version: 5.5.1
Qt theme: fusion

Build Info:
Build date: 2016-08-15 00:07:05 UTC
Sources date: 20160814
Revision number: git-6933-g65f77c7
System name: Linux-4.4.0-34-generic
CPU name: x86_64
Build command: /usr/bin/cmake
Build flags: 
   MANDIR=share/man
   CMAKE_INSTALL_PREFIX=/usr
   LIB_SUFFIX=/x86_64-linux-gnu
   Threads=POSIX
Compiler name: /usr/bin/c++
Compiler flags: N/A
Qt version: 5.5.1
Features: IRC, Debug, Crypt, SSL, IfAddr, IPC, OSS, Transparency, Webkit, Perl, Python, Enchant, Qt5, KVS
```

**Comment by un1versal** [comment]
> un1versal commented on 31 Oct 2015 • edited

On Window list treeview select any channel and right click and select undock.

ON Single monitor setup: the channel window should now be ontop of KVIrc program.
On Multimonitor setup (where KVIrc is on monitor 2) : the channel window is thrown to monitor 1

Issue 1) Multimonitor the undocked window shouldn't really be thrown to monitor 1 if KVIrc is running on Monitor 2

Now dock the Window list treeview select the undocked channel and right click and select dock.

issue 2) You cannot select the undocked channel on Window list treeview
~~Issue 3) Navigation with mouse scroll or ctrl+tab is broken~~ **fixed**
Issue 4) Right click over the undocked channel and watch how a different channel is selected instead

Issue 2/4 probably related to window focus and maybe even expected behaviour, but feels clumsy and confusing.

```
KVIrc 4.3.2 'Aria' Linux Architecture: x86_64

Build Date: 2015-10-31 07:10:14 UTC
Sources Date: 20151030
Revision Number: git-5613-gb8c034d
System Name: Linux-3.13.0-62-generic
CPU Name: x86_64
Build Command: /usr/bin/cmake
Build Flags: 
   MANDIR=man
   CMAKE_INSTALL_PREFIX=/usr
   LIB_SUFFIX=/x86_64-linux-gnu
   Threads=POSIX
Compiler Name: /usr/bin/c++
Compiler Flags: N/A
```

---

Merged #1703 into #2109 on Thu  1 Sep 08:59:38 BST 2016

**Comment by IceN9ne** [comment]
When attempting to click on an undocked window in the tree window list, you get this debug output.  Same if you try to `ctrl+tab` or `ctrl+shift+tab` to select it.

```
QStackedWidget::setCurrentWidget: widget 0x7fbc28008840 not contained in stack
```

I wonder if this was an issue before MDI was removed.
